### PR TITLE
fix: database-backed secret resolver

### LINKED ISSUE #1887: Add a database-backed secret resolver (basically identical to the config one, so consider refactoring that)
Currently we have an ASM resolver + provider in production, which results in all operations hitting ASM, and FTL in production being restricted to a single secret source. By having a database resolver, we can reduce the calls to ASM, and support multiple providers (eg. a local Linux keychain for single-host deploys, inline (needs additional encryption), etc.)

### PR DESCRIPTION
**Pull request by safeer**
Fixes #1887

### CODE REVIEW

File: backend/controller/sql/schema/001_init.sql [hunk 1]
```diff
@@ -518,4 +518,14 @@ CREATE TABLE module_configuration
     UNIQUE (module, name)
 );
 
+CREATE TABLE module_secrets
+(
+    id         BIGINT NOT NULL GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT (NOW() AT TIME ZONE 'utc'),
+    module     TEXT,  -- If NULL, configuration is global.
+    name       TEXT   NOT NULL,
+    value      JSONB  NOT NULL,
```
Comment (alecthomas) [inline backend/controller/sql/schema/001_init.sql:527]: This should be a URL for a provider, not the actual value.

File: backend/protos/xyz/block/ftl/v1/ftl.proto [hunk 2]
```diff
@@ -458,6 +458,9 @@ enum SecretProvider {
 
   // Store a secret in the AWS Secrets Manager.
   SECRET_ASM = 4;
+
+  // Use the database as a secret store.
+  SECRET_DB = 5;
```
Comment (alecthomas) [inline backend/protos/xyz/block/ftl/v1/ftl.proto:463]: I think there's been a misunderstanding. We only want the DB to be used as a Router, not a Provider. The Router just stores a URI that is then used to route to the actual provider, and the provider retrieves it.

Comment (safeer) [reply backend/protos/xyz/block/ftl/v1/ftl.proto:463]: Ahhh that makes sense! I thought the db was replacing the ASM store for some use cases like ftl box. On it.

File: common/configuration/dal/dal.go [hunk 3]
```diff
@@ -45,3 +46,35 @@ func (d *DAL) ListModuleConfiguration(ctx context.Context) ([]sql.ModuleConfigur
...
+
+func (d *DAL) UnsetModuleSecret(ctx context.Context, module optional.Option[string], name string) error {
+	err := d.db.UnsetModuleSecret(ctx, module, name)
+	if err != nil {
+		return fmt.Errorf("could not unset secret: %w", dalerrs.TranslatePGError(err))
+	}
+	return nil
+}
+
+func (d *DAL) ListModuleSecrets(ctx context.Context) ([]sql.ModuleSecret, error) {
```
Comment (alecthomas) [inline common/configuration/dal/dal.go:74]: The DAL shouldn't return types from `sql.` directly, that's breaking abstractions. If you want the exact same type though, just do `type ModuleSecret sql.ModuleSecret` and return that.

**Review by alecthomas — APPROVED** [review]
Apologies for that delay!

**Pull request merged**